src/flatpickr.js

```diff
@@ -1801,8 +1801,9 @@ function Flatpickr(element, config) {
 				.map(dObj => self.formatDate(dObj, self.config.altFormat))
 				.join(joinChar);
 		}
-
-		triggerEvent("ValueUpdate");
+		if (triggerChange !== false) {
```

Comment:
@zerkalica While `onChange` may cause `onValueUpdate`, `onValueUpdate` and `onChange` are separate events. As long as you supply `triggerChange=false` to `setDate()`, there won't be recursion or other problems :+1: